target_chains/sui/contracts/sources/pyth.move

```diff
@@ -392,16 +497,41 @@ module pyth::pyth_tests{
         verified_vaas
     }
 
+    // get_verified_vaa_from_accumulator_message parses the accumulator message up until the vaa, then
+    // parses the vaa, yielding a verified wormhole::vaa::VAA object
+    fun get_verified_vaa_from_accumulator_message(worm_state: &WormState, accumulator_message: vector<u8>, clock: &Clock): VAA {
+        let _PYTHNET_ACCUMULATOR_UPDATE_MAGIC: u64 = 1347305813;
+
+        let cursor = cursor::new(accumulator_message);
+        let header: u32 = deserialize::deserialize_u32(&mut cursor);
+        assert!((header as u64) == _PYTHNET_ACCUMULATOR_UPDATE_MAGIC, 0);
+        let major = deserialize::deserialize_u8(&mut cursor);
+        assert!(major == 1, 0);
+        let _minor = deserialize::deserialize_u8(&mut cursor);
+
+        let trailing_size = deserialize::deserialize_u8(&mut cursor);
+        deserialize::deserialize_vector(&mut cursor, (trailing_size as u64));
+
+        let proof_type = deserialize::deserialize_u8(&mut cursor);
+        assert!(proof_type == 0, 0);
+
+        let vaa_size = deserialize::deserialize_u16(&mut cursor);
+        let vaa = deserialize::deserialize_vector(&mut cursor, (vaa_size as u64));
+        cursor::take_rest(cursor);
+        vaa::parse_and_verify(worm_state, vaa, clock)
+    }
+
```

Comment:
How the users are going to do this from the client side? how should we separate it? I don't know whether it's possible or not, but maybe it's good if we have a public function that takes out the vaa bytes and then users can pass that object to wormhole. If that be possible we don't need to put this on the test

Reply:
We also need to add it to the readme for the users.

Reply:
We'll just need a typescript function that extracts the vaa bytes from an accumulator message. That's the only extra step.